### PR TITLE
fix: Fix broken nn.Flatten doc link in buildmodel_tutorial

### LINKED ISSUE #3948: Feedback about Build the Neural Network
There is the following issue on this page: https://docs.pytorch.org/tutorials/beginner/basics/buildmodel_tutorial.html#model-layers

Please, fix the incorrect reference on nn.Flatten.
Path: Build the Neural Network -> Model Layers -> nn.Flatten

Incorrect ref.: https://docs.pytorch.org/docs/stable/generated/torch.nn.Flatten.html
Correct ref.: https://docs.pytorch.org/docs/stable/generated/torch.nn.modules.flatten.Flatten.html

### PR DESCRIPTION
**Pull request by chinu0609**
Fixes #3948

Description

Update the nn.Flatten documentation link in beginner_source/basics/buildmodel_tutorial.py. The previous URL (https://pytorch.org/docs/stable/generated/torch.nn.Flatten.html) returns 404. Replace with the correct stable reference: https://docs.pytorch.org/docs/stable/generated/torch.nn.modules.flatten.Flatten.html.

Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #3948")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by chinu0609** [comment]
hi @albanD please review and merge

**Review by albanD — APPROVED** [review]
Thanks
